modules/dnn/test/test_layers_1d.cpp

```diff
@@ -480,6 +480,116 @@ INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_Scatter_Test, Combine(
 
 
 
+typedef testing::TestWithParam<tuple<std::vector<int>, std::string>> Layer_Reduce_Test;
+TEST_P(Layer_Reduce_Test, Accuracy_01D)
+{
+    auto reduceOperation = [](const cv::Mat& input, const std::string& operation, int axis) -> cv::Mat {
+        // Initialize result matrix
+        cv::Mat result;
+        if (shape(input).size() == 0 || shape(input).size() == 1){
+            result = cv::Mat(shape(input).size(), shape(input).data(), CV_32F);
+        } else {
+            if (axis == 0) {
+                result = cv::Mat::zeros(1, input.cols, CV_32F);
+            } else {
+                result = cv::Mat::zeros(input.rows, 1, CV_32F);
+            }
+        }
+
+        auto process_value = [&](float& res, float value, bool is_first) {
+            if (operation == "max") {
+                res = is_first ? value : std::max(res, value);
+            } else if (operation == "min") {
+                res = is_first ? value : std::min(res, value);
+            } else {
+                if (is_first) {
+                    if (operation == "sum" || operation == "l1" || operation == "l2"
+                        || operation == "sum_square" || operation == "mean" || operation == "log_sum"
+                        || operation == "log_sum_exp") res = 0;
+                    else if (operation == "prod") res = 1;
+                }
+
+                if (operation == "sum" || operation == "mean") res += value;
+                else if (operation == "sum_square") {
+                    if (shape(input).size() == 2 && shape(input)[0] == 1)
+                        res += value;
```

Comment:
Please clarify this condition. For example, `{1, 4}` input, axis=1 should be reduced to `{1x1}`:
 `[[a, b, c, d]]` -> `[[a*a + b*b + c*c + d*d]]`